### ISSUE TITLE
Add Variable Data Source Support

### ISSUE THREAD

**Issue by andgorrie** [scalyr/scalyr-grafana-datasource-plugin#108]
Please add the ability to use a DataSet data source as the source for a variable.  In my use case I want all distinct values for a specific Field to be used to populate a Query type variable.  Power Query format would be best, as then I could do some fairly advanced manipulation as well if I like.

**Comment by jmakar-scalyr** [comment]
This is supported (though as a facet query):
![Screenshot 2023-03-23 at 2 50 39 PM](https://user-images.githubusercontent.com/103437771/227335304-e0f8401e-0334-4ea3-87a3-c8ebbc14bb83.png)

Adding support for variables via power queries is a great idea, will add to the backlog

**Comment by andgorrie** [comment]
How do I filter with a facet query?  I got it to work returning a field but I can't figure out how to filter it to only the specific log I want.

**Comment by jmakar-scalyr** [comment]
It's a kludge, but currently the Regex field could be used to do this, for example above it could become something like `/^"(.*cortex.*)"$/`

**Comment by andgorrie** [comment]
I don't think that works for my use case.  I have multiple logs with the field 'project_id' and they are all integers, and I only want the 'project_id' from a specific log, so no way to sort the results, I need to filter the query. I assume this isn't currently possible?  I wasn't sure if there was a syntax to get to the filter in a FacetQuery.
